workflow/rules/common.smk

```diff
@@ -53,49 +53,47 @@ def get_trimmed(wildcards):
     # single end sample
     return expand("results/trimmed/{sample}-{unit}.fastq.gz", **wildcards)
 
+def is_activated(config_element):
+    return config_element['activate'] in {"true","True"}
+
 def get_bootstrap_plots(model, gene_list=None):
     """Dynamically determine which transcripts to plot based on
        checkpoint output."""
-    def inner(wildcards):
-        transcripts = dict()
-        genes = set()
-        # Obtain results from the sleuth_diffexp checkpoint.
-        # This happens dynamically after the checkpoint is completed, and
-        # is skipped automatically before completion.
-        results = pd.read_csv(
-            checkpoints.sleuth_diffexp.get(model=model).output.transcripts, sep="\t").dropna()
-        # group transcripts by gene
-        if gene_list is None:
-            results = results[results.qval <= config["bootstrap_plots"]["FDR"]][:config["bootstrap_plots"]["top_n"]] 
-            genes = set(results.ext_gene)
-        else:
-            genes = set(gene_list)
-            genes.add("Custom")
-        for g in genes:
-            if not pd.isnull(g):
-                valid = results.ext_gene == g
-                trx = set(results[valid].target_id)
-                transcripts[g] = trx
-        # Require the respective output from the plot_bootstrap rule.
-        return ["results/plots/bootstrap/{gene}/{gene}.{transcript}.{model}.bootstrap.pdf".format(gene=g, transcript=t, model=model)
-                for g, ts in transcripts.items()
-                for t in ts]
-    return inner
+    transcripts = dict()
+    genes = set()
+    # Obtain results from the sleuth_diffexp checkpoint.
+    # This happens dynamically after the checkpoint is completed, and
+    # is skipped automatically before completion.
+    results = pd.read_csv(
+        checkpoints.sleuth_diffexp.get(model=model).output.transcripts, sep="\t").dropna()
+    # group transcripts by gene
+    if gene_list is None:
+        results = results[results.qval <= config["bootstrap_plots"]["FDR"]][:config["bootstrap_plots"]["top_n"]]
+        genes = set(results.ext_gene)
+    else:
+        genes = set(gene_list)
+        genes.add("Custom")
+    for g in genes:
+        if not pd.isnull(g):
+            valid = results.ext_gene == g
+            trx = set(results[valid].target_id)
+            transcripts[g] = trx
+    # Require the respective output from the plot_bootstrap rule.
+    return ["results/plots/bootstrap/{gene}/{gene}.{transcript}.{model}.bootstrap.pdf".format(gene=g, transcript=t, model=model)
+            for g, ts in transcripts.items()
+            for t in ts]
```

Comment:
No inner anymore?

Reply:
Nope, this is now called in an input function to the `rule all` in the `Snakefile`, but in a `for` loop without list comprehension. Not as elegant as a `list comprehension`, but I find it more readable. But have a look here:

https://github.com/snakemake-workflows/rna-seq-kallisto-sleuth/pull/6/files#diff-54ba7003a36532ba119a71460aca1b42R87-R90